docs/advanced-features/static-html-export.md

```diff
@@ -35,24 +35,23 @@ For that you may want to update the scripts in your `package.json` like this:
 }
 ```
 
-And run it at once with:
+And run it with:
 
 ```bash
 npm run build
 ```
 
 Then you'll have a static version of your app in the `out` directory.
 
-By default `next export` doesn't require any configuration. It will generate a default `exportPathMap` with routes for the pages inside the `pages` directory.
-
-> To learn more about `exportPathMap` please visit the [documentation for the `exportPathMap` API](/docs/api-reference/next.config.js/exportPathMap.md).
+By default `next export` doesn't require any configuration. It will output a static HTML file for each page in your `pages` directory (except for [dynamic routes](/docs/routing/dynamic-routes.md), where it will call [`getStaticPaths`](/docs/basic-features/data-fetching.md#getstaticpaths-static-generation) and generate one or more pages based on the result). For more advanced scenarios, you can define a parameter called [`exportPathMap`](/docs/api-reference/next.config.js/exportPathMap.md) in your [`next.config.js`](/docs/api-reference/next.config.js/introduction.md) file to configure exactly which pages will be generated.
 
 ## Deployment
 
 You can read about deploying your Next.js application in the [deployment section](/docs/deployment.md).
 
 ## Caveats
 
-- With `next export`, we build an HTML version of your app. At export time we will run the [`getInitialProps`](/docs/api-reference/data-fetching/getInitialProps.md) in your pages. The `req` and `res` fields of the [`context`](/docs/api-reference/data-fetching/getInitialProps.md#context-object) object will be empty objects during export as there is no server running.
+- With `next export`, we build an HTML version of your app. At export time, we call [`getStaticProps`](/docs/basic-features/data-fetching.md#getstaticprops-static-generation) for each page that exports it, and pass the result to the page's component. If you're using the older [`getInitialProps`](/docs/api-reference/data-fetching/getInitialProps.md) API instead of `getStaticProps`, the `req` and `res` fields of the [`context`](/docs/api-reference/data-fetching/getInitialProps.md#context-object) object will be empty objects, since during export there is no server running.
```

Comment:
Combining `getStaticPaths` with `getInitialProps` and `next export` does not work for me as described here. `next export` gives me this message:

```Error: getStaticPaths was added without a getStaticProps in /classes/[filter]. Without getStaticProps, getStaticPaths does nothing
Error: Export encountered errors
```.

Once I replace `getInitialProps` with `getStaticProps`, `next export` completes.

Reply:
You're right, I guess the documentation even as I've amended it doesn't fully explain all the caveats of continuing to use `getInitialProps` -- I'll add another point.